browser-interface/packages/entryPoints/index.ts

```diff
@@ -25,137 +21,164 @@ import type { RootState } from 'shared/store/rootTypes'
+          // TODO: do we need `initSession`? Can't it be just a fork at the begining of the main saga?
+          await setupHomeAndInitialTeleport()
+          store.dispatch(initSession())
 
-    // initInternal must be called asynchronously, _after_ returning
-    async function initInternal() {
-      // load homepoint
-      const homePoint: string = await getFromPersistentStorage(homePointKey)
-      if (homePoint) {
-        store.dispatch(sendHomeScene(homePoint))
-      }
-
-      const urlPosition = getInitialPositionFromUrl()
-
-      // teleport to initial location
-      if (urlPosition) {
-        // 1. by URL
-        const { x, y } = urlPosition
-        store.dispatch(teleportToAction({ position: gridToWorld(x, y) }))
-      } else if (homePoint && !HAS_INITIAL_POSITION_MARK) {
-        // 2. by homepoint
-        const { x, y } = parseParcelPosition(homePoint)
-        store.dispatch(teleportToAction({ position: gridToWorld(x, y) }))
-      } else {
-        // 3. fallback to 0,0
-        const { x, y } = { x: 0, y: 0 } // '0,0'
-        store.dispatch(teleportToAction({ position: gridToWorld(x, y) }))
-      }
-
-      // Initializes the Session Saga
-      store.dispatch(initSession())
-
-      await Promise.all([initializeUnity(options.rendererOptions), loadWebsiteSystems(options.kernelOptions)])
-    }
+          await Promise.all([initializeUnity(options.rendererOptions), loadWebsiteSystems(options.kernelOptions)])
 
-    setTimeout(
-      () =>
-        initInternal().catch((err) => {
+        } catch (err: any) {
           BringDownClientAndReportFatalError(err, ErrorContext.WEBSITE_INIT)
-        }),
-      0
+        }
+      }, 0
     )
 
     return {
-      authenticate(provider: any, isGuest: boolean) {
-        if (!provider) {
-          throw new Error('A provider must be provided')
-        }
-        if (typeof provider === 'string') {
-          if (provider.startsWith('ws:') || provider.startsWith('wss:')) {
-            provider = new WebSocketProvider(provider)
-          } else {
-            throw new Error('Text provider can only be WebSocket')
-          }
-        }
-        return authenticateWhenItsReady(provider, isGuest)
-      },
+      authenticate: authenticateCallback,
       on: globalObservable.on.bind(globalObservable),
       version: 'mockedversion',
-      // this method is used for auto-login
-      async hasStoredSession(address: string, networkId: number) {
-        const existingSession = await retrieveLastSessionByAddress(address)
-        if (!existingSession) {
-          return { result: false }
-        }
+      hasStoredSession
+    }
+  }
+}
+
+function ensureHLSCapability() {
+  if (!Hls || !Hls.isSupported) {
+    throw new Error('HTTP Live Streaming did not load')
+  }
+
+  if (!Hls.isSupported()) {
+    throw new Error('HTTP Live Streaming is not supported in your browser')
+  }
```

Comment:
`Hls.isSupported()` is a function according Hls.js.

So this code should be like:
```suggestion
  if (!Hls) {
    throw new Error('HTTP Live Streaming did not load')
  }

  if (!Hls.isSupported()) {
    throw new Error('HTTP Live Streaming is not supported in your browser')
  }
```

I didn't find Hls.isSupported as a variable.

Reply:
I believe this is testing if the function exists so as to not call `undefined`